src/CarbonAware.DataSources/CarbonAware.DataSources.ElectricityMaps/src/Client/ElectricityMapsClient.cs

```diff
@@ -20,13 +19,15 @@ internal class ElectricityMapsClient : IElectricityMapsClient
     private ElectricityMapsClientConfiguration _configuration => this._configurationMonitor.CurrentValue;
     private readonly ILogger<ElectricityMapsClient> _log;
     private readonly Lazy<Task<Dictionary<string, ZoneData>>> _zonesAllowed;
+    private readonly bool _isFreeTrialUser;
 
     public ElectricityMapsClient(IHttpClientFactory factory, IOptionsMonitor<ElectricityMapsClientConfiguration> monitor, ILogger<ElectricityMapsClient> log)
     {
         _client = factory.CreateClient(IElectricityMapsClient.NamedClient);
         _configurationMonitor = monitor;
         _log = log;
         _configuration.Validate();
+        _isFreeTrialUser = this._configuration.BaseUrl == BaseUrls.TrialBaseUrl;
```

Comment:
Should we use `.equals` instead of `==` for string comparisons? Wonder if `==` compares by reference instead of value like java

Reply:
@pritipath, It seems [correct](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/equality-operators#string-equality)